nifi-extension-bundles/nifi-github-bundle/nifi-github-extensions/src/main/java/org/apache/nifi/github/GitHubFlowRegistryClient.java

```diff
@@ -0,0 +1,633 @@
+
+        final String version = flowVersionLocation.getVersion();
+        final String filePath = getSnapshotFilePath(flowVersionLocation);
+
+        final InputStream inputStream = repositoryClient.getContentFromCommit(filePath, version);
+        final RegisteredFlowSnapshot flowSnapshot = getSnapshot(inputStream);
+        populateFlowAndSnapshotMetadata(flowSnapshot, flowVersionLocation);
+
+        // populate values that aren't store in GitHub
+        flowSnapshot.getSnapshotMetadata().setVersion(version);
+        flowSnapshot.getSnapshotMetadata().setBranch(flowVersionLocation.getBranch());
+        flowSnapshot.getFlow().setBranch(flowVersionLocation.getBranch());
+
+        // populate outgoing bucket references
+        updateBucketReferences(flowSnapshot, flowVersionLocation.getBucketId());
+
+        // determine if the version is the "latest" version by comparing to the response of getLatestVersion
+        final String latestVersion = getLatestVersion(context, flowVersionLocation).orElse(null);
+        flowSnapshot.setLatest(version.equals(latestVersion));
+
+        return flowSnapshot;
+    }
+
+    @Override
+    public RegisteredFlowSnapshot registerFlowSnapshot(final FlowRegistryClientConfigurationContext context, final RegisteredFlowSnapshot flowSnapshot, final RegisterAction action)
+            throws FlowRegistryException, IOException {
+        final GitHubRepositoryClient repositoryClient = getRepositoryClient(context);
+        final RegisteredFlowSnapshotMetadata snapshotMetadata = flowSnapshot.getSnapshotMetadata();
+
+        final String branch = snapshotMetadata.getBranch();
+        final FlowLocation flowLocation = new FlowLocation(snapshotMetadata.getBranch(), snapshotMetadata.getBucketIdentifier(), snapshotMetadata.getFlowIdentifier());
+        final String filePath = getSnapshotFilePath(flowLocation);
+        final String previousSha = repositoryClient.getContentSha(filePath, branch).orElse(null);
+
+        final String snapshotComments = snapshotMetadata.getComments();
+        final String commitMessage = StringUtils.isBlank(snapshotComments) ? DEFAULT_FLOW_SNAPSHOT_COMMIT_MESSAGE : snapshotComments;
+
+        final RegisteredFlowSnapshot existingSnapshot = getSnapshot(filePath, branch);
+        populateFlowAndSnapshotMetadata(existingSnapshot, flowLocation);
+
+        final RegisteredFlow existingFlow = existingSnapshot.getFlow();
+        existingFlow.setBranch(null);
+        flowSnapshot.setFlow(existingFlow);
+
+        // Clear values we don't want stored in the json in GitHub
+        flowSnapshot.setBucket(null);
+        flowSnapshot.getSnapshotMetadata().setBucketIdentifier(null);
+        flowSnapshot.getSnapshotMetadata().setBranch(null);
+        flowSnapshot.getSnapshotMetadata().setVersion(null);
+        flowSnapshot.getSnapshotMetadata().setComments(null);
+        flowSnapshot.getSnapshotMetadata().setTimestamp(0);
```

Comment:
Is not timestemp we want to store? I think alternatively we can go with the timestamp of the commit itself, but as far as I found, this is set only in case of `getFlowVersions` not at the other methods where we try to acquire flow or version (every `RegisteredFlowSnapshot` contains the same metadata type as well)

Reply:
The reason for clearing these values is to reduce the differences someone would see in a diff for a pull request. It seemed meaningless to see the timestamp change in a pull request, so I was trying to narrow it down to only the real changes to the flow. Also, if you were looking at the file in GitHub you can see the timestamps of the commits, no need to see it in the json. 

NiFi does not use the timestamp inside the overall snapshot or the flow object for anything. It only uses the timestamps when listings versions. We could try to populate it on the way out, but from a quick look it will require making another request to GitHub, which I'm trying to limit.